execution-engine/src/fs.rs

```diff
@@ -334,20 +334,28 @@ impl FileSystem {
     // Operations on the filesystem. Rust style implementation of WASI API
     ////////////////////////////////////////////////////////////////////////////
 
-    /// The stub implementation of `clock_res_get`. Return unsupported error `NoSys`.
+    /// Get clock resolution from platform services. Return error `NoSys` in case of failure.
     #[inline]
-    pub(crate) fn clock_res_get(&self, _clock_id: ClockId) -> FileSystemResult<Timestamp> {
-        Err(ErrNo::NoSys)
+    pub(crate) fn clock_res_get(&self, clock_id: ClockId) -> FileSystemResult<Timestamp> {
+        let clock_id = u32::from(clock_id) as u8;
```

Comment:
What should happen if `clock_id` is not in the range `[0..255)`?  Where does this restriction come from, anyway?

Reply:
I think `ClockId` is effectively  `u8` https://github.com/oasislabs/wasi-types/blob/3414ddf62499e793d72710a4866ec9312ce0a035/src/lib.rs#L34. It should be a safe conversion here.

Reply:
Thanks, I'll mark this as resolved.